From eef03385338dc2b40b2bd5ed86d6c6150bcbe677 Mon Sep 17 00:00:00 2001
From: Beomjoong Yun <bj.yun@lge.com>
Date: Wed, 9 Jun 2021 15:41:43 +0900
Subject: [PATCH] Revert "Don't hide the inputMethod when finishing the edit"

This reverts commit 6da66ff611902d8c4d485568d746f49c69f1330f.
It is to keep the existing behavior of single-line TextInput that hides
the input method on the enter key.
---
Upstream-Status: Inappropriate [webOS specific]

 src/quick/items/qquicktextinput.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/quick/items/qquicktextinput.cpp b/src/quick/items/qquicktextinput.cpp
index 42720c6753..7f45e938b6 100644
--- a/src/quick/items/qquicktextinput.cpp
+++ b/src/quick/items/qquicktextinput.cpp
@@ -4424,6 +4424,8 @@ void QQuickTextInputPrivate::processKeyEvent(QKeyEvent* event)
 
             QInputMethod *inputMethod = QGuiApplication::inputMethod();
             inputMethod->commit();
+            if (!(q->inputMethodHints() & Qt::ImhMultiLine))
+                inputMethod->hide();
 
             if (activeFocus) {
                 // If we lost focus after hiding the virtual keyboard, we've already emitted
